src/stacked_borrows/diagnostics.rs

```diff
@@ -125,74 +116,17 @@ impl AllocHistory {
                 })
             });
 
-        if let SbTag::Tagged(_) = tag {
-            let get_matching = |events: &[Event]| {
-                events.iter().rev().find_map(|event| {
-                    if event.tag == tag { Some((event.range, event.span.data())) } else { None }
-                })
-            };
-            Some(TagHistory::Tagged {
-                tag,
-                created: get_matching(&self.creations)?,
-                invalidated: get_matching(&self.invalidations),
-                protected,
-            })
-        } else {
-            let mut created_time = 0;
-            // Find the most recently created tag that satsfies this offset
-            let recently_created = self.creations.iter().rev().find_map(|event| {
-                if event.tag == tag && offset >= event.range.start && offset < event.range.end() {
-                    created_time = event.time;
-                    Some((event.range, event.span.data()))
-                } else {
-                    None
-                }
-            });
-
-            // Find a different recently created tag that satisfies this whole operation, predates
-            // the recently created tag, and has a different span.
-            // We're trying to make a guess at which span the user wanted to provide the tag that
-            // they're using.
-            let matching_created = recently_created.and_then(|(_created_range, created_span)| {
-                self.creations.iter().rev().find_map(|event| {
-                    if event.tag == tag
-                        && alloc_range.start >= event.range.start
-                        && alloc_range.end() <= event.range.end()
-                        && event.span.data() != created_span
-                        && event.time != created_time
-                    {
-                        Some((event.range, event.span.data()))
-                    } else {
-                        None
-                    }
-                })
-            });
-
-            // Find the most recent invalidation of this tag which post-dates the creation
-            let recently_invalidated = recently_created.and_then(|_| {
-                self.invalidations
-                    .iter()
-                    .rev()
-                    .take_while(|event| event.time > created_time)
-                    .find_map(|event| {
-                        if event.tag == tag
-                            && offset >= event.range.start
-                            && offset < event.range.end()
-                        {
-                            Some((event.range, event.span.data()))
-                        } else {
-                            None
-                        }
-                    })
-            });
-
-            Some(TagHistory::Untagged {
-                recently_created,
-                matching_created,
-                recently_invalidated,
-                protected,
```

Comment:
@saethlin a lot of your code turned into dead code here...

Reply:
That's probably a good thing. This code implements a bunch of best-effort heuristics which are hard to explain to a new user.

I haven't looked into the state of the new diagnostics yet. I wouldn't be surprised if we need to add in a new branch somewhere for permissive provenance, but this code going away on its own does not concern me.